### ISSUE TITLE
How to set up your terminal for dbt

### ISSUE THREAD

**Issue by infused-kim** [dbt-labs/dbt-technical-blog-writing#85]
**What's your key point?**
An introduction to [fzf-dbt](https://github.com/Infused-Insight/fzf-dbt), a tool that allows you to dynamically select dbt models.

Check out the github page for a demo video:
https://github.com/Infused-Insight/fzf-dbt

**Prior art:**
No

**Link to notes / outline / draft:**
https://docs.google.com/document/d/1m-tmeqFMYjEyJ1yWmwINT_fuxU_KW-FUx1fDTror_Ok/edit#

**Estimated first draft date:**
Already done

**Any open questions / requests for help from the group?**

**Comment by jasnonaz** [comment]
Hi there - first of all I want to thank you for building this. It's is a really cool tool and I'm excited to dive in / for more people to get a chance to check this out. 

I think that this post is not a great fit for the a Dev Blog post in it's current form, but if we can make a more general post on node selection best practices or other hands on, tutorial type content we'd be happy to revisit this. In the meantime feel free to post on the [Discourse](https://discourse.getdbt.com/)

**Comment by infused-kim** [comment]
No problem, I totally understand.

Would it help if it was more practical. Perhaps a post that is more focused on instructions on how to set up your terminal to make working with dbt better?

**Comment by jasnonaz** [comment]
Oooh yes a post on how to set up your terminal to make working with dbt better would _definitely_ be a good fit for the blog!

**Comment by infused-kim** [comment]
Fantastic, what do you think of a broad outline like this:

- Why and how to install dbt in a pythen virtual environment
- How to add shell completions to dbt
- How to add interactive search to dbt

This would probably work well with another serious of articles that describe how to set up a user's favorite code editor for dbt.

I imagine there could be an article for VSCode and another one for Atom. And all of them could interlink.

What do you think of that?

**Comment by jasnonaz** [comment]
That sounds like a great idea - I wonder if there are any other things we would want to add there. Feel free to ask how people do this in #analytics-craft on the slack.

**Comment by infused-kim** [comment]
Great. I have asked for more ideas here:
https://getdbt.slack.com/archives/C02FH9AD85T/p1643732821330979

But in the meantime I’ll get started on the sections we agreed on so far. 

We can always expand it with more tips from the community once we have an initial draft.

**Comment by infused-kim** [comment]
Actually, while thinking about it there are a few more things that I have done to improve my terminal experience that I think would be good to include:

1. Why and how to install dbt in a python virtual environment
2. How to add shell completions to dbt
3. How to add interactive search to dbt
4. **How to replace common unix tools with modern alternatives**
5. **How to customize your shell prompt for an improved development experience**

**Comment by infused-kim** [comment]
Ok, here is a first draft:
https://docs.google.com/document/d/12BjJX7fxKBRmZc3e4Rsg5cDYU7HIoXElEX4ej-uyjVQ/edit?usp=sharing

## Outline Changes
I have remembered a few more terminal tips that I have been using and that I think will be very valuable for the dbt community.

And I took out the python virtual env part, because the article is already at 2,000 words and 15 pages.

But if you think python environments are important for this article, I would still be happy to write that part up too.

## Voice
I have chosen a more casual and banter-y voice for this article.

But if you think something more serious or corporate-y would be a better fit, I can easily change it.

Check it out and let me know what you think @jasnonaz.

**Comment by jasnonaz** [comment]
Reopening this under the new scope about how to use dbt in the terminal

**Comment by gwenwindflower** [comment]
hi @infused-kim - there's so much great stuff in here! 

your article highlighted a new concern for us though which is wanting to make sure we've published a standardized guide on a topic _before_ publishing a more opinionated, advanced, or specific piece.

we imaging we'd want to publish this as a 'Here's how I setup my terminal for advanced tooling and fuzzy finding of models' vs 'Here's how you should set up your terminal'. The lack of our guide being published makes this distinction blurry, whereas if the guide was out we could easily ensure that this reads as one way of customizing your shell (which to reiterate, we _very_ much want to publish this, and that kind of thing more generally).

An exercise we need to do internally to save people getting "ahead of the roadmap" so to speak is getting this all mapped out here publicly so people can see the full vision for the docs and slot pieces in for transparently.

Overall this article is looking very cool, but we've got a few things to sort out interally before we can publish -- we will get back to you with next steps in the next couple days, just wanted to give you a heads up on where we're at - thanks!

**Comment by infused-kim** [comment]
Hi @gwenwindflower,

thank you for the update. I completely understand the concern and look forward to the next steps.

In the meantime, I think it's worth considering that `How to set up your terminal for dbt` was the original idea for the article.

But if you take a look at the headline I used in the [google doc](https://docs.google.com/document/d/12BjJX7fxKBRmZc3e4Rsg5cDYU7HIoXElEX4ej-uyjVQ/edit#heading=h.eb579cce9b9t), it's `5 Tips To Take Your dbt Terminal Experience To The Next Level`.

I think by using that headline, we can probably already address a large portion of your concern.

With that headline I doubt anyone would confuse this with an "official" way to set up their terminal. Instead it will just be seen as 5 tips that a user can consider implementing alongside with the standardized guide you will publish in the future.

I think it also helps that all five tips are all optional "nice to haves" and not necessary for the usage of dbt.

**Comment by infused-kim** [comment]
Hi @gwenwindflower. I don't mean to be pushy, but I just wanted to check in and see if there is anything I can do to move this forward?

**Comment by infused-kim** [comment]
Hi @gwenwindflower  and @jasnonaz.

I just wanted to check in and see if you have made any decisions about this article.

If you have decided internally that you don't want to publish it, I would completely understand it. If you do want to publish it, but need more more time or want to change something, I am also happy to work with you on this.

I'd just like to get some clarity on where we are at since it's been in limbo for over three weeks now.

**Comment by jasnonaz** [comment]
Hi @infused-kim - I'm really sorry about the delay here, appreciate you bearing with us. In the interest of making it so we aren't keeping you waiting any longer, I think we are going to shelf this post as a developer blog piece right now - we think it would get confusing for folks looking to do terminal setup while we don't have any guidance out there. We really, really appreciate your contributions to the community.

In the interest of getting an area to link to do discuss dbt-fzf, perhaps you'd be interested in posting the original writeup you were considering (how and why to use dbt-fzf in the "[i-made-this](https://github.com/dbt-labs/docs.getdbt.com/discussions/categories/i-made-this)" section of the new docs Discussion area. This will hopefully give folks a chance to discover what you've built!

**Comment by infused-kim** [comment]
> I think we are going to shelf this post as a developer blog piece right now - we think it would get confusing for folks looking to do terminal setup while we don't have any guidance out there. 

Thank you for letting me know @jasnonaz. 

Is there any way you would reconsider? I think a simple solution is to simply not to use the title `How to set up your terminal` and instead use the headline I put in the article: `5 Tips To Take Your dbt Terminal Experience To The Next Level`.

With this title I don't think there will be any confusion whether this is official guidance or not.

And the content matches that expectation too, because it doesn't detail anything about dbt installation or python environments. Instead every tip I present is a completely optional thing that could make working with dbt more pleasant.

What do you think?

**Comment by jasnonaz** [comment]
I think not right now for the Developer Blog _but_ that's not the only way to share things with the community! We definitely want you to be able to get this information out, if you do end up publishing elsewhere definitely share on #i-made-this in the slack and that way folks will still get a chance to benefit from these tips.

**Comment by infused-kim** [comment]
Ok, I understand. Thank you for considering it!
